### DISCUSSION TITLE
WebDAV/Nextcloud : Images from nextcloud not getting indexed

### DISCUSSION THREAD

**Discussion by xssChauhan** [photoprism/photoprism#1083] [Q&A] [answered]
I have a nextcloud server, and I am trying out photoprism.

Photoprism was able to successfully index the pictures on my local drive.

I also setup syncing with nextcloud, as described in the [docs](https://docs.photoprism.org/user-guide/settings/sync/).

Photoprism does not index the folder I pointed to on my nextcloud installation.  Is this supported, or I misunderstood the purpose of sync?

**Comment by lastzero** [comment, marked as answer]
It won't index your files directly on the Nextcloud server without downloading them as this would be much too slow, and we need the original file anyway to create thumbnails. Downloading them may take a while. This is not done or triggered when you start indexing in Library.

**Comment by PiotrEsse** [comment]
I also have issues with Nextcloud sync. I am able to correctly add nextcloud ccount. Than I am trying to activate Sync - everything is going smoothly. I am open sync options and enable sync - than I see folder list on nextcloud and I am picking one sucessfully  - so Photoprosm is able to sucessfully connect to NC and browse folder list. 
After saving sync options nothing happens. PP dont start downloading and indexing pictures. 
As a side note - I was trying to connect to 3 separate NC instances and in one case it worked but with NC installation method provided by https://www.c-rieger.de/ in both cases PP cant download and sync files (but it can connect to NC w/o issues)
I dont see any errors in logs or maybe I look in wrong places - can You help?

**Comment by lastzero** [comment]
You may need to wait a little, depending on the worker interval configured.

**Comment by PiotrEsse** [comment]
I am waiting days. trying to configure it few times already, everytime waiting many hours or even 1-2 day. ;-)
I think theres something preventing proper download and indexing. 
As I said - one NC worked flawlessly (plain Apache). Other NC are using Nginx - whatever it changes.
I was trying to do this on 3 completely different Photoprism installations (different VM, IP's etc). Same issue.

**Comment by lastzero** [comment]
Nginx doesn't work as chunked requests are not supported in combination with PHP. Bad choice for webdav.

**Comment by PiotrEsse** [comment]
LastZero - thank You, finally my few months mystery has been solved ;-)
Ok, it would be adviceble to put notice somewhere in Sync window to warn users about Nginx. Ill create an issue.

**Reply by matthijsbro** [reply]
I'm a bit confused about this. Will it never work with NC on nginx? Or just be slow? cause I'm waiting now and wonder if I should just trigger that download / index somewhere.

**Reply by lastzero** [reply]
Nginx with PHP is not fully webdav compatible. Can't fix it.

**Reply by matthijsbro** [reply]
thanks. I does work, it just takes quite a while. But it starts indexing the files.

**Comment by lastzero** [comment]
Actually they should put this notice in Nextcloud as it is not fully WebDAV compatible like that 😉

**Comment by lastzero** [comment]
See https://trac.cyberduck.io/wiki/help/en/howto/mount/issues/fastcgi

Since this seems about downloading, not uploading, there might be another and a similar issue. Or it's a simple timeout as getting the file list is taking too long.
